x/campaign/keeper/msg_server_update_total_supply.go

```diff
@@ -35,6 +37,15 @@ func (k msgServer) UpdateTotalSupply(goCtx context.Context, msg *types.MsgUpdate
 		return nil, sdkerrors.Wrapf(types.ErrMainnetInitialized, "%d", msg.CampaignID)
 	}
 
+	// Validate provided totalSupply
+	totalSupplyRange := k.TotalSupplyRange(ctx)
+	if err := types.ValidateTotalSupply(msg.TotalSupplyUpdate, totalSupplyRange); err != nil {
+		if errors.Is(err, types.ErrInvalidSupplyRange) {
```

Comment:
This branch here is not covered from tests (same in `msg_create_campaign`) since it would require to create a state where the range params are invalid. I assumed this would be fine

Reply:
Yes, critical errors are generally (except some exception) not tested simply because can't create a state where those are triggered unless we do some hacks in the store which is not in the idea of message tests